frontend/packages/kubevirt-plugin/src/selectors/vm-like/nextRunChanges.tsx

```diff
@@ -0,0 +1,106 @@
+import * as React from 'react';
+import { VMWrapper } from '../../k8s/wrapper/vm/vm-wrapper';
+import { VMIWrapper } from '../../k8s/wrapper/vm/vmi-wrapper';
+import * as _ from 'lodash';
+import { BootableDeviceType } from '../../types/types';
+import { VMKind, VMIKind } from '../../types/vm';
+import { Alert, AlertVariant } from '@patternfly/react-core';
+import { MODAL_RESTART_IS_REQUIRED } from '../../strings/vm/status';
+import { IsPendingChange } from '../../components/vms/types';
+
+import './nextRunChanges.scss';
+import { getBootableDevicesInOrder } from '../vm/devices';
+import { getVMIBootableDevicesInOrder } from '../vmi/devices';
+import { mapCDsToSource } from '../../components/modals/cdrom-vm-modal/helpers';
+import { K8sResourceKind } from '../../../../../public/module/k8s/types';
+
+export const isFlavorChanged = (vm: VMWrapper, vmi: VMIWrapper): boolean => {
+  if (!vm || !vmi) {
+    return false;
+  }
+
+  return (
+    vm.getFlavor() !== vmi.getFlavor() ||
+    !_.isEqual(vm.getMemory(), vmi.getMemory()) ||
+    !_.isEqual(vm.getCPU(), vmi.getCPU())
+  );
+};
+
+export const isCDROMChanged = (
+  vm: VMWrapper,
+  vmi: VMIWrapper,
+  dataVolumes: K8sResourceKind[],
+): boolean => {
+  if (!vm || !vmi) {
+    return false;
+  }
+
+  const vmiCDs = mapCDsToSource(vmi.getCDROMs(), vmi.asResource(), dataVolumes);
+  const vmCDs = mapCDsToSource(vm.getCDROMs(), vm.asResource());
+
+  const vmCDNames = Object.keys(vmCDs);
+  const vmiCDNames = Object.keys(vmiCDs);
+  if (vmCDNames.length !== vmiCDNames.length) {
+    return true;
+  }
+
+  return !vmCDNames.every(
+    (name) =>
+      vmiCDNames.includes(name) &&
+      _.isEqual(_.omit(vmiCDs[name], 'cdrom.readonly', 'cdrom.tray'), vmCDs[name]),
```

Comment:
why are we omitting fields here?

Reply:
Because there is a constant difference between VMI disks and VM disks.
here we are searching if a cdrom has changed - if there is a difference between the cdrom configuration in the VMI and the VM.
In the VMI CDROM there is 2 additional fields in the `cdrom` key which are: `readonly` and `tray`.
without omitting them `_.isEqual` will always return `false`.